### PR TITLE
[PREVIEW] DIV-3203 Adultery admission question

### PR DESCRIPTION
**Pull request by alexwakeman**
# Description

Introducing a new screen flow for Adultery cases.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested manually, with unit tests and with functional tests using the new mock backend service.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: middleware/petitionMiddleware.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const CaptureCaseAndPin = require('steps/capture-case-and-pin/CaptureCaseAndPin.
 const InvalidCaseState = require('steps/invalid-case-state/InvalidCaseState.step');
 
 const AOS_STARTED = 'AosStarted';
+const AWAITING_AOS = 'AwaitingAos';
```
Comment (hosainnet) [inline middleware/petitionMiddleware.js]: This should be `AosAwaiting`. I think mocks flying have this incorrectly.
https://github.com/hmcts/div-case-maintenance-service/blob/91b4c50c3a77884c7bf1db5cc1f33f12f6b274ad/src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/domain/model/CaseState.java

File: middleware/petitionMiddleware.js [hunk 2]
```diff
@@ -30,7 +31,7 @@ const loadMiniPetition = (req, res, next) => {
     .then(response => {
       if (response.statusCode === SUCCESS) {
         const caseState = response.body.state;
-        if (caseState !== AOS_STARTED) {
+        if (caseState !== AOS_STARTED && caseState !== AWAITING_AOS) {
```
Comment (hosainnet) [inline middleware/petitionMiddleware.js]: As far as I know, if the case has been linked it'll always be started. Unless there is some weird edge case? 
@mthava can confirm?

File: package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "start": "NODE_PATH=. node server.js",
     "setup": "NODE_PATH=.",
+    "devMocks": "NODE_PATH=.; yarn mocks | yarn dev",
```
Comment (hosainnet) [inline package.json]: Not sure the `;` is necessary. Also should we make this the default behaviour of `dev` as it won't really function without mocks? And maybe update the README.md

File: test/unit/steps/confirmDefence.test.js [hunk 4]
```diff
@@ -4,6 +4,8 @@ const Jurisdiction = require('steps/jurisdiction/Jurisdiction.step');
 const ChooseAResponse = require('steps/choose-a-response/ChooseAResponse.step');
 const confirmDefenceContent = require('steps/confirm-defence/ConfirmDefence.content');
 const idam = require('services/idam');
+// eslint-disable-next-line
+const adulteryCase = require('mocks/services/case-orchestration/retrieve-aos-case/mock-adultery-case');
```
Comment (hosainnet) [inline test/unit/steps/confirmDefence.test.js]: This doesn't seem to be used

File: package.json (hunk 3 above)
Comment (alexwakeman) [reply package.json]: Possibly - however I'm leaning towards leaving it out in the end - it works better in separate terminal instances

File: middleware/petitionMiddleware.js (hunk 2 above)
Comment (alexwakeman) [reply middleware/petitionMiddleware.js]: Started means the user is picking it back up I assumed - awaiting AOS means the respondent has never logged in before - I could be wrong though...

**Pull request merged**